TestChangesLoopingWhenLowSequence: Sequence 6 never showed up! (unit tests against couchbase bucket)

### Sync Gateway version
Sg commit: f37de25ce1b0cc8bc9646d6266bb965d6ca88190 (equivalent to master commit: https://github.com/couchbase/sync_gateway/commit/a1cc33c0e4f56875d36580d895e7939b04bc89bf)

### Log output
```
2017-06-22T11:57:15.395-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{_sync:user:}") count=2
2017-06-22T11:57:16.187-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{_sync:user:bernard}") count=3
2017-06-22T11:57:16.187-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{*}") count=4
2017-06-22T11:57:16.188-07:00 Debug: Waiting for sequence: 6
--- FAIL: TestChangesLoopingWhenLowSequence (8.92s)
panic: changeCache: Sequence 6 never showed up! [recovered]
	panic: changeCache: Sequence 6 never showed up!

goroutine 6 [running]:
panic(0xb13d60, 0xc420456a90)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc42008a780)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0xb13d60, 0xc420456a90)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/couchbase/sync_gateway/db.(*changeCache).waitForSequenceWithMissing(0xc42020e5a0, 0x6)
	/var/jenkins/workspace/build-sync-gateway/godeps/src/github.com/couchbase/sync_gateway/db/change_cache.go:316 +0x19e
github.com/couchbase/sync_gateway/db.(*DatabaseContext).WaitForSequenceWithMissing(0xc420098500, 0x6, 0x1, 0x1)
	/var/jenkins/workspace/build-sync-gateway/godeps/src/github.com/couchbase/sync_gateway/db/changes.go:727 +0xe8
github.com/couchbase/sync_gateway/rest.TestChangesLoopingWhenLowSequence(0xc42008a780)
	/var/jenkins/workspace/build-sync-gateway/godeps/src/github.com/couchbase/sync_gateway/rest/changes_api_test.go:560 +0x470
```

### Steps to reproduce
1. Run this jenkins job: http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway/103/parameters/

--- a/db/change_cache_test.go
+++ b/db/change_cache_test.go
@@ -213,6 +213,10 @@ func WriteUserDirect(db *Database, username string, sequence uint64) {
 
 func WriteDirectWithKey(db *Database, key string, channelArray []string, sequence uint64) {
 
+	if base.TestUseXattrs() {
+		panic(fmt.Sprintf("WriteDirectWithKey() cannot be used in tests that are xattr enabled"))
+	}
+
 	rev := "1-a"
 	chanMap := make(map[string]*channels.ChannelRemoval, 10)
 
@@ -233,6 +237,10 @@ func WriteDirectWithKey(db *Database, key string, channelArray []string, sequenc
 // simulating out-of-order arrivals on the tap feed using walrus.
 
 func WriteDirectWithChannelGrant(db *Database, channelArray []string, sequence uint64, username string, channelGrantArray []string) {
+
+	if base.TestUseXattrs() {
+		panic(fmt.Sprintf("WriteDirectWithKey() cannot be used in tests that are xattr enabled"))
+	}
 
 	docId := fmt.Sprintf("doc-%v", sequence)
 	rev := "1-a"

--- a/manifest/default.xml
+++ b/manifest/default.xml
@@ -33,7 +33,7 @@
 
   
   <!-- Sync Gateway Accel-->
-  <project groups="notdefault,sg-accel" name="sync-gateway-accel" path="godeps/src/github.com/couchbaselabs/sync-gateway-accel" remote="couchbaselabs_private" revision="b5185b1407e6893e3a871e1f4b76eed448ef6870"/>
+  <project groups="notdefault,sg-accel" name="sync-gateway-accel" path="godeps/src/github.com/couchbaselabs/sync-gateway-accel" remote="couchbaselabs_private" revision="db103c3931e396be787c77aa12a70d4b2ac9a3b8"/>
 
   <!-- Dependencies specific to Sync Gateway Accel-->
   <project groups="notdefault,sg-accel" name="cbgt" path="godeps/src/github.com/couchbase/cbgt" remote="couchbase" revision="4aaaca921a7ef64900f3b569c33ad87e9e8df065"/>

--- a/rest/changes_api_test.go
+++ b/rest/changes_api_test.go
@@ -1101,6 +1101,10 @@ func WriteDirect(testDb *db.DatabaseContext, channelArray []string, sequence uin
 
 func WriteDirectWithKey(testDb *db.DatabaseContext, key string, channelArray []string, sequence uint64) {
 
+	if base.TestUseXattrs() {
+		panic(fmt.Sprintf("WriteDirectWithKey() cannot be used in tests that are xattr enabled"))
+	}
+
 	rev := "1-a"
 	chanMap := make(map[string]*channels.ChannelRemoval, 10)
 

--- a/rest/changes_api_test.go
+++ b/rest/changes_api_test.go
@@ -532,6 +532,10 @@ func postChangesAdminChannelGrant(t *testing.T, it indexTester) {
 // longpoll as well as clients doing repeated one-off changes requests - see #1309)
 func TestChangesLoopingWhenLowSequence(t *testing.T) {
 
+	if base.TestUseXattrs() {
+		t.Skip("This test cannot run in xattr mode until WriteDirect() is updated.  See https://github.com/couchbase/sync_gateway/issues/2666#issuecomment-311183219")
+	}
+
 	pendingMaxWait := uint32(5)
 	maxNum := 50
 	skippedMaxWait := uint32(120000)

